meta-client/src/test/java/org/apache/helix/metaclient/impl/zk/TestZkMetaClient.java

```diff
@@ -334,6 +320,36 @@ public void handleDirectChildChange(String key) throws Exception {
     }
   }
 
+  @Test
+  public void testConnectStateChangeListener() throws Exception {
+    final String basePath = "/TestZkMetaClient_testConnectionStateChangeListener";
+    try (ZkMetaClient<String> zkMetaClient = createZkMetaClient()) {
+      CountDownLatch countDownLatch = new CountDownLatch(1);
+      final MetaClientInterface.ConnectState[] connectState =
+          new MetaClientInterface.ConnectState[2];
+      ConnectStateChangeListener listener = new ConnectStateChangeListener() {
+        @Override
+        public void handleConnectStateChanged(MetaClientInterface.ConnectState prevState,
+            MetaClientInterface.ConnectState currentState) throws Exception {
+          System.out.println("from: " + prevState + " to State: " + currentState);
+          connectState[0] = prevState;
+          connectState[1] = currentState;
+          countDownLatch.countDown();
+        }
+
+        @Override
+        public void handleConnectionEstablishmentError(Throwable error) throws Exception {
+
+        }
+      };
+      Assert.assertTrue(zkMetaClient.subscribeStateChanges(listener));
+      zkMetaClient.connect();
+      countDownLatch.await(5000, TimeUnit.SECONDS);
+      Assert.assertEquals(connectState[0], MetaClientInterface.ConnectState.NOT_CONNECTED);
+      Assert.assertEquals(connectState[1], MetaClientInterface.ConnectState.CONNECTED);
+    }
```

Comment:
Nice to have: can we have more test cases on other state changes?

Reply:
Will add more test case when connection lost and reconnect is implemented.